include/boost/geometry/io/wkt/write.hpp

```diff
@@ -62,22 +64,94 @@ namespace boost { namespace geometry
 namespace detail { namespace wkt
 {
 
+template <typename Output>
+struct output_formatter
+{
+    template <typename T>
+    static void append(Output& out, T const& v, int = -1)
+    {
+        out << v;
+    }
+};
+
+template <>
+struct output_formatter<std::string>
+{
+    template <typename T>
+    static void append(std::string& out, T const& val, int significant_digits = -1)
+    {
+        BOOST_STATIC_ASSERT((boost::is_arithmetic<T>::value));
+        std::stringstream ss;
+        if(std::numeric_limits<T>::is_specialized)
+        {
+            if(significant_digits < 0)
+                ss.precision(std::numeric_limits<T>::digits10);
+            else
+                ss.precision(significant_digits);
```

Comment:
The precision is set only if `numeric_limits` are specialized for type `T`.

Reply:
@awulkiew Do you mean this?

```cpp
if(significant_digits < 0 && std::numeric_limits<T>::digits10 > 0) // or std::numeric_limits<T>::is_specialized as equivalent
    ss.precision(std::numeric_limits<T>::digits10);
```

Reply:
@mloskot I mean that if a user passes `significant_digits` it means he/she wants it to be set in all cases, i.e.:

    if (significant_digits >= 0)
        ss.precision(significant_digits);
    else if (std::numeric_limits<T>::is_specialized)
        ss.precision(std::numeric_limits<T>::digits10);
        
or am I missing something?

Reply:
>  wants it to be set in all cases

...regardless if `std::numeric_limits<T>` is specialized or not.

Yes, correct.